.github/workflows/build.yml

```diff
@@ -195,6 +195,36 @@ jobs:
           name: coverage
           path: ./coverage
 
+  test-freebsd:
+    runs-on: macos-10.15
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      - name: Test in FreeBSD
+        env:
+          TMPDIR: ${{ runner.temp }}
+          SKIP_INTEGRATION_TESTS: 1
+        id: test
+        uses: vmactions/freebsd-vm@v0.1.5
```

Comment:
I'm not familiar with this action; is this one we need to pin by commit @crazy-max ? https://github.com/vmactions/freebsd-vm

Reply:
Looks like the `vmactions/freebsd-vm` action downloads a random freebsd image from a GitHub repo: https://github.com/vmactions/freebsd-vm/blob/45665d548dfeadbbebb5bb7971150211525fee46/index.js#L90. Don't really like it.

Would be better to use vagrant like: https://github.com/tonistiigi/fsutil/blob/251a862f39279cbec4c35fbb6f79e1a6d04421ad/.github/workflows/ci.yml#L76-L97

Reply:
I'm not sure either. The Vagrant approach has proven unreliable, so I decided to look for alternatives.

Reply:
Unreliable as in sporadic HTTP 429s

Reply:
Yes indeed to avoid rate-limit issue we have to add a scheduled event in the workflow (every 6 days) so cache is not evicted. See https://github.com/tonistiigi/fsutil/pull/112

Reply:
But as BuildKit repo is pretty active I think it's not necessary to add this event.

Reply:
Yes, it's a bit of a pain to run anything other than "the usual" on GitHub actions 😔

Reply:
We'll get the native runners eventually :smile: 

Let's stick with Vagrant for now.

Reply:
An alternative would be to use https://cirrus-ci.com/, which is what I'm using for runj.